include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -1004,7 +1004,7 @@ namespace VectorTools
       Assert (dof.get_fe().degree == fe_degree,
               ExcDimensionMismatch(fe_degree, dof.get_fe().degree));
 
-      typedef MatrixFreeOperators::MassOperator<dim, fe_degree, 1, Number> MatrixType;
+      typedef MatrixFreeOperators::MassOperator<dim, fe_degree, n_q_points_1d, 1, Number> MatrixType;
```

Comment:
@kronbichler : it is this function, which uses `n_q_points_1d` for both matrix-free mass matrix and RHS calculation from `Table<>` quadrature data.